trin-beacon/src/storage.rs

```diff
@@ -375,14 +356,13 @@ impl BeaconStorage {
 
     pub fn store(
         &mut self,
-        key: &impl OverlayContentKey,
+        key: &BeaconContentKey,
```

Comment:
`Self::Key` can't be used here. It can only be used inside `impl ContentStore for BeaconStorage {`.

While we have to define `type Key =` (in order to implement `ContentStore` trait, I removed `Self::Key` from function signature because:
- if you are looking at function only, it's more explicit and readable
- it's more consistent with the rest of the file

Also applied to history and state.